### PR TITLE
Fix NetStandard issue in System.IO.FileSystem.AccessControl in 5.0

### PR DESCRIPTION
**Pull request by carlossanlop**
Related change in 3.1: https://github.com/dotnet/corefx/pull/42501

Fixes: https://github.com/dotnet/corefx/issues/42483

### CODE REVIEW

File: src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj [hunk 1]
```diff
@@ -1,10 +1,16 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AllowUnsafeBlocks Condition="'$(TargetsWindows)' == 'true'">true</AllowUnsafeBlocks>
+    <DefineConstants Condition="'$(TargetsNetCoreApp)' != 'true'">$(DefineConstants);INTERNAL_NULLABLE_ATTRIBUTES</DefineConstants>
```
Comment (safern) [inline src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: This shouldn't be needed anymore, you can just add a `<Nullable>annotations</Nullable>` property and you'll get this constant and the `NullableAttributes.cs` file imported when the configuration needs it:

https://github.com/dotnet/corefx/blob/cb5be3e9c76c337cb1752b206a50ffc71947aba1/src/Directory.Build.targets#L12-L21

File: src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
     <IsPartialFacadeAssembly Condition="'$(TargetsNetFx)' == 'true'">true</IsPartialFacadeAssembly>
     <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsWindows)' != 'true'">SR.PlatformNotSupported_AccessControl</GeneratePlatformNotSupportedAssemblyMessage>
     <Configurations>net461-Windows_NT-Debug;net461-Windows_NT-Release;netcoreapp-Windows_NT-Debug;netcoreapp-Windows_NT-Release;netfx-Windows_NT-Debug;netfx-Windows_NT-Release;netstandard2.0-Debug;netstandard2.0-Release;netstandard2.0-Windows_NT-Debug;netstandard2.0-Windows_NT-Release</Configurations>
   </PropertyGroup>
+  <!-- Handle nullability/GC in NetFX/NetStandard -->
+  <ItemGroup Condition="'$(TargetsNetCoreApp)' != 'true'">
+    <Compile Include="$(CommonPath)\CoreLib\System\Diagnostics\CodeAnalysis\NullableAttributes.cs" Link="System\Diagnostics\CodeAnalysis\NullableAttributes.cs" />
```
Comment (safern) [inline src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: You shouldn't need to import this file if you add `<Nullable>annotations</Nullable>` as suggested above.

File: src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj [hunk 3]
```diff
@@ -1,10 +1,16 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AllowUnsafeBlocks Condition="'$(TargetsWindows)' == 'true'">true</AllowUnsafeBlocks>
+    <DefineConstants Condition="'$(TargetsNetCoreApp)' != 'true'">$(DefineConstants);INTERNAL_NULLABLE_ATTRIBUTES</DefineConstants>
     <IsPartialFacadeAssembly Condition="'$(TargetsNetFx)' == 'true'">true</IsPartialFacadeAssembly>
     <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsWindows)' != 'true'">SR.PlatformNotSupported_AccessControl</GeneratePlatformNotSupportedAssemblyMessage>
     <Configurations>net461-Windows_NT-Debug;net461-Windows_NT-Release;netcoreapp-Windows_NT-Debug;netcoreapp-Windows_NT-Release;netfx-Windows_NT-Debug;netfx-Windows_NT-Release;netstandard2.0-Debug;netstandard2.0-Release;netstandard2.0-Windows_NT-Debug;netstandard2.0-Windows_NT-Release</Configurations>
   </PropertyGroup>
+  <!-- Handle nullability/GC in NetFX/NetStandard -->
```
Comment (safern) [inline src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: This comment is not necessary

File: src/Common/src/CoreLib/System/IO/Path.cs [hunk 4]
```diff
@@ -910,29 +910,21 @@ private static string GetRelativePath(string relativeTo, string path, StringComp
         /// <summary>
         /// Trims one trailing directory separator beyond the root of the path.
         /// </summary>
-        public static string TrimEndingDirectorySeparator(string path) =>
-            EndsInDirectorySeparator(path) && !PathInternal.IsRoot(path.AsSpan()) ?
-                path.Substring(0, path.Length - 1) :
-                path;
+        public static string TrimEndingDirectorySeparator(string path) => PathInternal.TrimEndingDirectorySeparator(path);
```
Comment (stephentoub) [inline src/Common/src/CoreLib/System/IO/Path.cs]: This is changing some files that are compiled into Corelib as well.  Have you validated that Corelib successfully builds if you copy all of these Common/src/CoreLib files over to your coreclr repo?

Comment (carlossanlop) [reply src/Common/src/CoreLib/System/IO/Path.cs]: @stephentoub I copied the changes for `Path.cs` and `PathInternal.cs` to coreclr and it finished building successfully:

image

**Pull request merged**